sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1298,6 +1298,105 @@ int gnrc_netif_ipv6_add_prefix(gnrc_netif_t *netif,
 out:
     return res;
 }
+
+#if IS_USED(MODULE_GNRC_NETIF_BUS)
+static bool _has_global_addr(gnrc_netif_t *netif)
+{
+    ipv6_addr_t ipv6_addrs[CONFIG_GNRC_NETIF_IPV6_ADDRS_NUMOF];
+    int res = gnrc_netapi_get(netif->pid, NETOPT_IPV6_ADDR, 0, ipv6_addrs,
+                              sizeof(ipv6_addrs));
```

Comment:
Since this is within `gnrc_netif.c` I think it is fine to use `gnrc_netif_acquire()`/`gnrc_netif_release()` in this function and access the IPv6 address array of the netif directly. Saves some cycles and memory.